packages/cadl-ranch-specs/http/client/structure/common/service.tsp

```diff
@@ -0,0 +1,60 @@
+import "@typespec/rest";
+import "@azure-tools/typespec-client-generator-core";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using Azure.ClientGenerator;
+using Azure.ClientGenerator.Core;
+
+@doc("""
+  Test that we can use @client and @operationGroup decorators to customize client side code structure, such as:
+  1. have everything as default.
+  2. to rename client or operation group
+  3. one client can have more than one operations groups
+  4. split one interface into two clients
+  5. have two clients with operations come from different interfaces
+  6. have two clients with a hierarchy relation.
+  """)
+@server(
+  "http://localhost:3000/client/structure/{client}",
+  "",
+  {
+    client: ClientTypeString,
```

Comment:
Why not `client: ClientType`? I assume `@knownValues` is an outdate way to define extensible enum.

If it is an enum, just use an enum. (playground is good, not sure how emitter works)

Reply:
changed to ClientType.